widgetssdk/src/main/java/com/glia/widgets/chat/adapter/ChatVisitorMessageAttachmentAdapter.java

```diff
@@ -0,0 +1,160 @@
+                if (imageView.getTag() == null || !imageView.getTag().equals(attachmentFile.getName())) {
+
+                    Glia.getCurrentEngagement().ifPresent(engagement -> engagement.fetchFile(attachmentFile.getId(),
+                            (inputStream, e) -> {
+                                Picasso picassoCustom
+                                        = new Picasso.Builder(itemView.getContext()).addRequestHandler(new RequestHandler() {
+                                    @Override
+                                    public boolean canHandleRequest(Request data) {
+                                        return true;
+                                    }
+
+                                    @Override
+                                    public Result load(Request request, int networkPolicy) throws IOException {
+                                        return new Result(inputStream, Picasso.LoadedFrom.NETWORK);
+                                    }
+                                }).build();
+                                picassoCustom.load(attachmentFile.getName()).resize(0, 720).into(imageView);
+                                imageView.setTag(attachmentFile.getName());
+                            }
+                    ));
+                }
+            }
+        }
+    }
+
+    static class FileViewHolder extends RecyclerView.ViewHolder {
+        public FileViewHolder(@NonNull View itemView) {
+            super(itemView);
+        }
+
+
+        public void onBind(AttachmentFile attachmentFile, ItemCallback callback) {
+            CardView extensionContainerView = itemView.findViewById(R.id.type_indicator_view);
+            TextView extensionTypeText = itemView.findViewById(R.id.type_indicator_text);
+
+            TextView titleText = itemView.findViewById(R.id.item_title);
+            TextView statusIndicator = itemView.findViewById(R.id.status_indicator);
+            LinearProgressIndicator progressIndicator = itemView.findViewById(R.id.progress_indicator);
+
+            itemView.setOnClickListener(
+                    view -> {
+                        itemView.setOnClickListener(null);
+                        statusIndicator.setText("Downloading file...");
+                        progressIndicator.setVisibility(View.VISIBLE);
+                        // TODO itemcallback.onDownloadFile(id)
+                    }
+            );
+
+            String name = attachmentFile.getName();
+            long byteSize = attachmentFile.getSize();
+
+            extensionContainerView.setCardBackgroundColor(ContextCompat.getColor(itemView.getContext(), R.color.glia_brand_primary_color));
+            titleText.setText(String.format("%s • %s", name, Formatter.formatFileSize(itemView.getContext(), byteSize)));
+            String extension = Utils.getExtensionByStringHandling(name).orElse("");
+            extensionTypeText.setText(extension);
+        }
+    }
+
+    private interface ItemCallback {
```

Comment:
is this needed?

Reply:
Will be needed for file download callback